src/main/java/practice/StreamPractice.java

```diff
@@ -1,29 +1,45 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
+    public static final String SPLIT = ",";
     /**
      * Given list of strings where each element contains 1+ numbers:
      * input = {"5,30,100", "0,22,7", ...}
      * return min integer value. One more thing - we're interested in even numbers.
      * If there is no needed data throw RuntimeException with message
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream().map(s -> s.split(SPLIT))
+                .flatMap(Arrays::stream)
+                .map(Integer::parseInt)
+                .filter(integer -> integer % 2 == 0)
+                .mapToInt(Integer::intValue)
+                .min().orElseThrow(() ->
+                        new RuntimeException("Can't get min value from list:" + numbers));
     }
-
     /**
      * Given a List of Integer numbers,
      * return the average of all odd numbers from the list or throw NoSuchElementException.
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
+
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(index -> index % 2 == 0 ? numbers.get(index) : numbers.get(index) - 1)
+                .filter(integer -> integer % 2 != 0)
+                .average()
+                .orElseThrow();
```

Comment:
```suggestion
                .getAsDoublbe();
```
As stated in Common Mistakes